evap/staff/tests/test_views.py

```diff
@@ -185,64 +188,136 @@ def setUpTestData(cls):
         baker.make(UserProfile, pk=4)
 
 
-class TestUserBulkDeleteView(WebTest):
-    url = '/staff/user/bulk_delete'
-    filename = os.path.join(settings.BASE_DIR, 'staff/fixtures/test_user_bulk_delete_file.txt')
+class TestUserBulkUpdateView(WebTest):
+    url = '/staff/user/bulk_update'
+    filename = os.path.join(settings.BASE_DIR, 'staff/fixtures/test_user_bulk_update_file.txt')
+    filename_random = os.path.join(settings.BASE_DIR, 'staff/fixtures/random.random')
+    filename_xls = os.path.join(settings.BASE_DIR, 'staff/fixtures/test_enrollment_data.xls')
 
     @classmethod
     def setUpTestData(cls):
-        baker.make(UserProfile, username='manager', groups=[Group.objects.get(name='Manager')])
+        cls.user = baker.make(UserProfile, username='manager', groups=[Group.objects.get(name='Manager')])
 
     def test_testrun_deletes_no_users(self):
         page = self.app.get(self.url, user='manager')
-        form = page.forms['user-bulk-delete-form']
+        form = page.forms['user-bulk-update-form']
 
         form['username_file'] = (self.filename,)
 
         baker.make(UserProfile, is_active=False)
-        users_before = UserProfile.objects.count()
+        users_before = set(UserProfile.objects.all())
 
         reply = form.submit(name='operation', value='test')
 
-        # Not getting redirected after.
         self.assertEqual(reply.status_code, 200)
         # No user got deleted.
-        self.assertEqual(users_before, UserProfile.objects.count())
+        self.assertEqual(users_before, set(UserProfile.objects.all()))
+
+        helper_delete_all_import_files(self.user.id)
+
+    @override_settings(INSTITUTION_EMAIL_DOMAINS=["institution.example.com", "internal.example.com"])
+    def test_multiple_email_matches_trigger_error(self):
+        baker.make(UserProfile, email='testremove@institution.example.com')
+        baker.make(UserProfile, email='testuser1@institution.example.com')
+
+        error_string = (
+            'Multiple users match the email testuser1@institution.example.com:'
+            + '<br />None None (testuser1@institution.example.com)'
+            + '<br />None None (testuser1@internal.example.com)'
+        )
+        button_substring = 'value="bulk_update"'
 
-    def test_deletes_users(self):
-        baker.make(UserProfile, username='testuser1')
-        baker.make(UserProfile, username='testuser2')
+        expected_users = set(UserProfile.objects.all())
```

Comment:
This variable seams not to be checked after `new_user` is added to it. Did you forgot to check it after the test run (which wouldn’t add that much value imo)? In the other case, can you move it down right before the assertion?

Reply:
Yup, I thought that testing after both test runs are done also implies that no user was lost in the middle, but you're right, making it explicit is a bit nicer. Added.